src/apps/create-patron-user-info/UserInfo.tsx

```diff
@@ -35,27 +35,44 @@ const UserInfo: FC<UserInfoProps> = ({ cpr }) => {
     phoneNumber: "",
     emailAddress: ""
   });
-
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [isSubmitError, setIsSubmitError] = useState<boolean>(false);
+  const getSubmitButtonText = () => {
+    if (isLoading) {
+      return t("createPatronButtonLoadingText");
+    }
+    if (isSubmitError) {
+      return t("createPatronButtonErrorText");
+    }
+    return t("createPatronConfirmButtonText");
+  };
   // Changes the patron object by key.
-  // So using the paramters 123 and "phoneNumber" would change the phoneNumber to 123.
+  // So using the parameters 123 and "phoneNumber" would change the phoneNumber to 123.
   const changePatron = (newValue: string | boolean, key: string) => {
-    // Deeeep copy
+    // Deep copy
     const copyUser = JSON.parse(JSON.stringify(patron));
     set(copyUser, key, newValue);
     setPatron(copyUser);
   };
 
   const handleSubmit = (e: FormEvent<HTMLFormElement>) => {
     e.preventDefault();
-    const { preferredPickupBranch, phoneNumber, emailAddress } = patron;
-    if (pin && preferredPickupBranch && phoneNumber && emailAddress) {
+    setIsLoading(true);
+    const { preferredPickupBranch, emailAddress } = patron;
+
+    if (pin && preferredPickupBranch && emailAddress) {
       mutate(
         {
           data: { cprNumber: cpr, patron, pincode: pin }
         },
         {
           onSuccess: () => {
+            setIsLoading(false);
             redirectTo(redirectOnUserCreatedUrl);
+          },
+          onError: () => {
+            setIsLoading(false);
+            setIsSubmitError(true);
           }
```

Comment:
I remember I created a whole system for a situation like this, where you have a submit button and then you get a feedback modal that can either fail or display a success message. I don't know if it would make sense to use it here. I also have PR coming with changes to this bit that forces the user to be logged out and log in again.

Reply:
In here we're not in a modal context.